bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/ControllerHandler.java

```diff
@@ -44,116 +47,163 @@
             }
-        } catch (CoolMasterClientError e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+        if (isConnected()) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
         }
     }
 
-    /*
-     * Return true if the client socket is connected.
+    /**
+     * Passively determine if the client socket appears to be connected, but do
+     * modify the connection state.
      *
-     * Use checkConnection() to probe if the coolmasternet is responding correctly,
-     * and try to re-establish the connection if possible.
+     * <p>
+     * Use {@link #checkConnection()} if active verification (and potential
+     * reconnection) of the CoolNetMaster connection is required.
      */
     public boolean isConnected() {
         synchronized (socketLock) {
-            Socket localSocket = socket;
-            return localSocket != null && localSocket.isConnected() && !localSocket.isClosed();
+            final Socket socket = this.socket;
+            return socket != null && socket.isConnected() && !socket.isClosed();
         }
     }
 
-    /*
-     * Send a particular ASCII command to the CoolMasterNet, and return the successful response as a string.
+    /**
+     * Send a specific ASCII I/F command to CoolMasterNet and return its response.
+     *
+     * <p>
+     * This method automatically acquires a connection.
      *
-     * If the "OK" prompt is not received then a CoolMasterClientError is thrown that contains whatever
-     * error message was printed by the CoolMasterNet.
+     * @return the server response to the command (never empty)
+     * @throws {@link IOException} if communications failed with the server
      */
-    @SuppressWarnings("resource")
-    public @Nullable String sendCommand(String command) throws CoolMasterClientError {
+    public String sendCommand(final String command) throws IOException {
         synchronized (socketLock) {
             checkConnection();
 
-            StringBuilder response = new StringBuilder();
+            final StringBuilder response = new StringBuilder();
             try {
-                Socket localSocket = socket;
-                if (localSocket == null || !isConnected()) {
+                final Socket socket = this.socket;
+                if (socket == null || !isConnected()) {
                     throw new CoolMasterClientError(String.format("No connection for sending command %s", command));
                 }
 
                 logger.trace("Sending command '{}'", command);
-                OutputStream out = localSocket.getOutputStream();
-                out.write(command.getBytes(StandardCharsets.US_ASCII));
+                OutputStream out = socket.getOutputStream();
+                out.write(command.getBytes(US_ASCII));
                 out.write(LF);
 
```

Comment:
It is good practice to make sure that you flush the output before you try to read a response.
```suggestion
                out.write(LF);
                out.flush();

```

Reply:
Done.